app/src/main/java/com/github/multimatum_team/multimatum/repository/FirebaseAuthRepository.kt

```diff
@@ -30,39 +54,15 @@ class FirebaseAuthRepository @Inject constructor() : AuthRepository {
      * Create an anonymous user so that the user can try the application before having to link it
      * to their Google account.
      */
-    private suspend fun anonymousSignIn(): AnonymousUser {
-        val authResult = auth.signInAnonymously().await()
-        if (authResult.user == null) {
-            throw RuntimeException("Failed to sign-in anonymously")
-        } else {
-            return AnonymousUser(authResult.user!!.uid)
-        }
-    }
-
-    /**
-     * Get the currently logged-in user.
-     * We guarantee that an user is always logged in at any point of the application.
-     * At the first execution of the application, an anonymous user is created, and logged into on
-     * subsequent executions, until the user chooses to link the anonymous account to an actual
-     * Google account (see SignInActivity).
-     */
-    override suspend fun getCurrentUser(): User {
-        lateinit var user: User
-        val firebaseUser = auth.currentUser
-        if (firebaseUser == null) {
-            // When the user was never logged-in (when the app is started the first time), we create
-            // an anonymous firebase account
-            Log.d(TAG, "First time log-in, waiting for anonymous sign-in...")
-            user = anonymousSignIn()
-            Log.d(TAG, "Successfully signed-in anonymously with user ID ${user.id}")
-        } else {
-            // Otherwise, we check whether the user has signed in using a real account or if they
-            // were signed in as an anonymous user
-            user = convertFirebaseUser(firebaseUser)
+    private suspend fun anonymousSignIn(): AnonymousUser =
+        when (val authResultUser = auth.signInAnonymously().await()?.user) {
+            null ->  throw RuntimeException("Failed to sign-in anonymously")
```

Comment:
It would probably be better to replace `RuntimeException` by a more precise type of exception.

Reply:
What do you suggest? This snippet was copied from the Firebase docs.

Reply:
`IllegalStateException`? The sign-in fails, which should not happen and now the repository is in a state that it cannot handle

Reply:
Looking at the documentation of `IllegalStateException`, it seems that it's rather thrown to signal that the method was called in an illegal state rather than the method leaving the program in an illegal state

Reply:
Indeed, ideally you could define a custom exception as an object but it's fine to keep it like that, the message does the job